base/services/tcpsvcs/log.c

```diff
@@ -130,7 +130,7 @@ LogToFile(LPCWSTR lpMsg,
 
         bRet = WriteFile(hLogFile,
                          lpFullMsg,
-                         wcslen(lpFullMsg) * sizeof(WCHAR),
+                         (DWORD)wcslen(lpFullMsg) * sizeof(WCHAR),
```

Comment:
Many of these casts are questionable. This one is an example that I consider outright wrong. You're silencing a legitimate warning about integer overflow without fixing the problem. I'd rather have the warning to help us in the future when we actually go and overflow-proof the codebase.

Reply:
Agreed with @ThFabba, cast should be the last resort solution, and not the first tried solution.
Blindly casting can lead to severe issues, or even security issues and should be avoided.
I take for example this commit: https://github.com/reactos/reactos/commit/371aba1abee3f872ee9f146015f9db59ec6297df (the root cause of this bug was a cast "to make MSVC build").

Reply:
I agree that these things are very unlikely to happen as accidents. Malicious actors tend to find a way to make them happen though, so it's a security concern. And I know that we only marginally care about security at this point -- which is why my recommendation was to just leave the warning be a warning.
That said, by all means feel free to fix it properly, whether that be an exception or silent truncation.